diff -u -rb libssh-0.6.4.org/cmake/Modules/DefineCompilerFlags.cmake libssh-0.6.4/cmake/Modules/DefineCompilerFlags.cmake
--- libssh-0.6.4.org/cmake/Modules/DefineCompilerFlags.cmake	2014-09-16 02:54:34.000000000 +0800
+++ libssh-0.6.4/cmake/Modules/DefineCompilerFlags.cmake	2015-05-03 16:52:33.000000000 +0800
@@ -10,7 +10,7 @@
     if (${CMAKE_C_COMPILER_ID} MATCHES "(GNU|Clang)")
 
         # add -Wconversion ?
-        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99 -pedantic -pedantic-errors")
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99")
         set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -Wshadow -Wmissing-prototypes -Wdeclaration-after-statement")
         set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wunused -Wfloat-equal -Wpointer-arith -Wwrite-strings -Wformat-security")
         set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wmissing-format-attribute")
