diff -u -r pulseaudio-5.0.org/shell-completion/pulseaudio-bash-completion.sh pulseaudio-5.0/shell-completion/pulseaudio-bash-completion.sh
--- pulseaudio-5.0.org/shell-completion/pulseaudio-bash-completion.sh	2014-01-23 22:57:55.000000000 +0400
+++ pulseaudio-5.0/shell-completion/pulseaudio-bash-completion.sh	2014-11-14 12:31:29.000000000 +0300
@@ -112,7 +112,7 @@
     local cur prev words cword preprev command
     local comps
     local flags='-h --help --version -s --server= --client-name='
-    local list_types='short sinks sources sink-inputs source outputs cards
+    local list_types='short sinks sources sink-inputs source-outputs cards
                     modules samples clients'
     local commands=(stat info list exit upload-sample play-sample remove-sample
                     load-module unload-module move-sink-input move-source-output
