Index: kmail-4.14.10/messageviewer/CMakeLists.txt
===================================================================
--- kmail-4.14.10.orig/messageviewer/CMakeLists.txt
+++ kmail-4.14.10/messageviewer/CMakeLists.txt
@@ -24,6 +24,7 @@ endif()
 
 configure_file(settings/messageviewer.kcfg.cmake ${CMAKE_CURRENT_BINARY_DIR}/messageviewer.kcfg)
 
+include(CheckIncludeFiles)
 check_include_files(sys/inotify.h SYS_INOTIFY_H_FOUND)
 macro_bool_to_01(SYS_INOTIFY_H_FOUND HAVE_SYS_INOTIFY_H)
 configure_file(config-messageviewer.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config-messageviewer.h)
