diff -u -r -b lensfun-0.3.0.org/libs/lensfun/lens.cpp lensfun-0.3.0/libs/lensfun/lens.cpp
--- lensfun-0.3.0.org/libs/lensfun/lens.cpp	2014-10-01 01:37:34.000000000 +0800
+++ lensfun-0.3.0/libs/lensfun/lens.cpp	2015-05-04 18:52:18.000000000 +0800
@@ -294,7 +294,7 @@
 
     if (!Model || !Mounts || CropFactor <= 0 ||
         MinFocal > MaxFocal || (MaxAperture && MinAperture > MaxAperture) ||
-        !isfinite (AspectRatio) || AspectRatio < 1)
+        !std::isfinite (AspectRatio) || AspectRatio < 1)
         return false;
 
     return true;
