diff -u -r -b kephal-4.11.7.org/libs/kephal/service/xrandr12/randroutput.cpp kephal-4.11.7/libs/kephal/service/xrandr12/randroutput.cpp
--- kephal-4.11.7.org/libs/kephal/service/xrandr12/randroutput.cpp	2015-02-27 08:18:57.000000000 +0800
+++ kephal-4.11.7/libs/kephal/service/xrandr12/randroutput.cpp	2015-05-04 12:42:45.000000000 +0800
@@ -227,7 +227,7 @@
 RandRMode RandROutput::mode() const
 {
     if (!isConnected())
-        return None;
+        return NULL;
 
     if (!m_crtc)
         return RandRMode();
