--- a/lib/gif_lib.h
+++ b/lib/gif_lib.h
@@ -20,6 +20,7 @@ extern "C" {
 
 #include <stddef.h>
 #include <stdbool.h>
+#include <stdlib.h>
 
 #define GIF_STAMP "GIFVER"          /* First chars in file - GIF stamp.  */
 #define GIF_STAMP_LEN sizeof(GIF_STAMP) - 1
@@ -244,9 +245,6 @@ extern ColorMapObject *GifUnionColorMap(
                                      GifPixelType ColorTransIn2[]);
 extern int GifBitSize(int n);
 
-extern void *
-reallocarray(void *optr, size_t nmemb, size_t size);
-
 /******************************************************************************
  Support for the in-core structures allocation (slurp mode).              
 ******************************************************************************/
