diff -u -r -b qt-everywhere-opensource-src-4.8.6.org/src/3rdparty/javascriptcore/JavaScriptCore/wtf/HashTable.h qt-everywhere-opensource-src-4.8.6/src/3rdparty/javascriptcore/JavaScriptCore/wtf/HashTable.h
--- qt-everywhere-opensource-src-4.8.6.org/src/3rdparty/javascriptcore/JavaScriptCore/wtf/HashTable.h	2014-04-11 02:37:12.000000000 +0800
+++ qt-everywhere-opensource-src-4.8.6/src/3rdparty/javascriptcore/JavaScriptCore/wtf/HashTable.h	2015-05-04 15:33:42.000000000 +0800
@@ -257,7 +257,7 @@
 
     using std::swap;
 
-#if !COMPILER(MSVC) && !OS(QNX) && !defined(_LIBCPP_VERSION)
+#if !COMPILER(MSVC) && !OS(QNX) && !defined(_LIBCPP_VERSION) && (!defined(__GNUC__) || (__GNUC__ < 5))
     // The Dinkumware C++ library (used by MSVC and QNX) and clang's libc++ have a swap for pairs defined.
 
     // swap pairs by component, in case of pair members that specialize swap
