diff -u -r -b oprofile-1.0.0.org/libutil++/op_bfd.cpp oprofile-1.0.0/libutil++/op_bfd.cpp
--- oprofile-1.0.0.org/libutil++/op_bfd.cpp	2014-09-12 22:39:45.000000000 +0800
+++ oprofile-1.0.0/libutil++/op_bfd.cpp	2015-05-04 19:15:45.000000000 +0800
@@ -535,7 +535,7 @@
 {
 	op_bfd_symbol const & sym = syms[sym_idx];
 
-	bool const verbose = cverb << (vbfd & vlevel1);
+	bool const verbose = bool(cverb << (vbfd & vlevel1));
 
 	if (anon_obj)
 		start = sym.vma();
