--- a/src/kbd.c
+++ b/src/kbd.c
@@ -383,7 +383,7 @@ KbdProc(DeviceIntPtr device, int what)
      */
     if (pInfo->fd >= 0) {
 	xf86FlushInput(pInfo->fd);
-	AddEnabledDevice(pInfo->fd);
+	xf86AddEnabledDevice(pInfo->fd);
     }
 
     device->public.on = TRUE;
@@ -397,7 +397,7 @@ KbdProc(DeviceIntPtr device, int what)
      * Restore original keyboard directness and translation.
      */
     if (pInfo->fd != -1)
-      RemoveEnabledDevice(pInfo->fd);
+      xf86RemoveEnabledDevice(pInfo->fd);
     pKbd->KbdOff(pInfo, what);
     device->public.on = FALSE;
     break;
