diff -u -r -b xapian-core-1.2.20.org/queryparser/queryparser_internal.cc xapian-core-1.2.20/queryparser/queryparser_internal.cc
--- xapian-core-1.2.20.org/queryparser/queryparser_internal.cc	2015-03-04 12:45:30.000000000 +0800
+++ xapian-core-1.2.20/queryparser/queryparser_internal.cc	2015-05-04 11:09:21.000000000 +0800
@@ -1636,7 +1636,7 @@
 #define BOOL_OP_TO_QUERY(E, A, OP, B, OP_TXT) \
     do {\
 	if (!A || !B) {\
-	    state->error = "Syntax: <expression> "OP_TXT" <expression>";\
+	    state->error = "Syntax: <expression> " OP_TXT " <expression>";\
 	    yy_parse_failed(yypParser);\
 	    return;\
 	}\
diff -u -r -b xapian-core-1.2.20.org/queryparser/queryparser.lemony xapian-core-1.2.20/queryparser/queryparser.lemony
--- xapian-core-1.2.20.org/queryparser/queryparser.lemony	2015-03-04 12:36:07.000000000 +0800
+++ xapian-core-1.2.20/queryparser/queryparser.lemony	2015-05-04 11:08:54.000000000 +0800
@@ -1627,7 +1627,7 @@
 #define BOOL_OP_TO_QUERY(E, A, OP, B, OP_TXT) \
     do {\
 	if (!A || !B) {\
-	    state->error = "Syntax: <expression> "OP_TXT" <expression>";\
+	    state->error = "Syntax: <expression> " OP_TXT " <expression>";\
 	    yy_parse_failed(yypParser);\
 	    return;\
 	}\
